### PR TITLE
WIP: migration from vcsim

### PR DESCRIPTION
**Pull request by rokkbert**
This is the scripts and config changes I made to migrate a simulated VM from vcsim.
It doesn't succeed at getting the disk. See MIGRATE_FROM_VCSIM.md for status and instructions to reproduce.
Any ideas and suggestions on how to further debug this are welcome.

### CODE REVIEW

File: vmware/vsphere_provider.yml [hunk 1]
```diff
@@ -26,4 +26,4 @@ spec:
   settings:
     vddkInitImage: 'quay.io/lrotenbe/vddk:7.0.3'
```
Comment (ahadas) [inline vmware/vsphere_provider.yml:27]: can we do such thing? need to check how CDI did this.. can it work without vddk? can we use the image CDI use, if they use vddk in their CI

File: build_and_setup_everything_bazel_manually.sh [hunk 2]
```diff
@@ -16,5 +16,7 @@
 
 . ./grant_permissions.sh
 
+kubectl patch --type=merge StorageProfile standard -p '{"spec":{"claimPropertySets":[{"accessModes":["ReadWriteOnce"],"volumeMode":"Filesystem"}]}}'
```
Comment (ahadas) [inline build_and_setup_everything_bazel_manually.sh:19]: we can drop that when #12 gets in, right?

Comment (rokkbert) [reply build_and_setup_everything_bazel_manually.sh:19]: Yes, it is the same as ovirt/patch.yml from that PR.

**Comment by eslutsky** [comment]
following https://github.com/vmware/govmomi/blob/master/Dockerfile.vcsim using podman-docker
```
docker push localhost:5001/vcsim
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Getting image source signatures
Error: trying to reuse blob sha256:eaad488b8d129de222b5965d2b7610ac1619ce4d7006f558c28d7164e5f5b145 at destination: pinging container registry localhost:5001: Get "https://localhost:5001/v2/": http: server gave HTTP response to HTTPS client
```

the fix was to add  to `/etc/containers/registries.conf`

```
[[registry]]
location="localhost:5001"
insecure=true
```

File: MIGRATE_FROM_VCSIM.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+This needs to be done only once. Get the
+[Dockerfile](https://github.com/vmware/govmomi/blob/master/Dockerfile.vcsim) for vcsim. 
+Change the last line to disable TLS:
+
+    CMD ["-l", "0.0.0.0:8989", "-tls=false"]
+
+Build the image: 
+
+    docker build -f Dockerfile.vcsim -t localhost:5001/vcsim .
```
Comment (eslutsky) [inline MIGRATE_FROM_VCSIM.md:23]: before running this build vcsim binary must downloaded  to the same dir from https://github.com/vmware/govmomi/blob/master/vcsim/README.md  OR built using [make](https://github.com/vmware/govmomi/blob/master/vcsim/Makefile)

File: vmware/setup.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
-kubectl apply -f ./vmware/vcsim_deployment.yml
+kubectl apply -f ./vmware/vcsim_deployment_http.yml
```
Comment (eslutsky) [inline vmware/setup.sh:1]: i would add this steps to make the proccess more automatic
```bash
wget https://raw.githubusercontent.com/vmware/govmomi/master/Dockerfile.vcsim
curl -L -o - https://github.com/vmware/govmomi/releases/latest/download/vcsim_$(uname -s)_$(uname -m).tar.gz | tar -xvzf - vcsim
docker build -f ./vmware/Dockerfile.vcsim -t localhost:5001/vcsim .
docker push localhost:5001/vcsim

```

Comment (mnecas) [reply vmware/setup.sh:1]: AFAIK the vcsim does not have any disk image in the container image. So we will need to build it with it.
So we can't use the Dockerfile from vcsim master.

File: vmware/vsphere_provider.yml (hunk 1 above)
Comment (ahadas) [reply vmware/vsphere_provider.yml:27]: yeah, my concern is about the licensing of vddk - it if permits publishing the image like that

File: vmware/setup.sh (hunk 4 above)
Comment (ahadas) [reply vmware/setup.sh:1]: I guess vcsim is pretty stable but still I wouldn't take the latest version on each run.. wouldn't it be better to build the container one in a while and push it to quay?

**Comment by eslutsky** [comment]
we no longer need this PR, it's handled by https://github.com/kubev2v/forkliftci/pull/25

**Comment by mnecas** [comment]
Moved to https://github.com/kubev2v/forkliftci/pull/25, closing.

**Pull request closed without merging**